### PR TITLE
Use createGoals on SDM instance

### PR DESCRIPTION
**Pull request by cdupuis**

### CODE REVIEW

File: lib/machine/springSupport.ts [hunk 1]
```diff
@@ -61,18 +51,44 @@ import {
...
-export function addSpringSupport(sdm: SoftwareDeliveryMachine): void {
+export const SpringGoalConfigurer: GoalConfigurer<SpringGoals> = async (sdm, goals) => {
 
-    autofix.with(springFormat(sdm.configuration));
+    sdm.addGeneratorCommand<SpringProjectCreationParameters>({
+        name: "create-spring",
+        intent: "create spring",
+        description: "Create a new Java Spring Boot REST service",
+        parameters: SpringProjectCreationParameterDefinitions,
+        startingPoint: GitHubRepoRef.from({ owner: "atomist-playground", repo: "spring-rest-seed", branch: "master" }),
```
Comment (lievendoclo) [inline lib/machine/springSupport.ts]: Perhaps use one of the atomist-seeds repos here?

Comment (cdupuis) [reply lib/machine/springSupport.ts]: That seed has some special stuff for playground. Also it is not relevant for this PR.

File: index.ts [hunk 2]
```diff
@@ -14,62 +14,96 @@
...
 
-            }
-            return config;
+    sdm.addExtensionPacks(
+        buildAwareCodeTransforms({
+            buildGoal: goals.build,
+            issueCreation: {
+                issueRouter: {
+                    raiseIssue: async () => { /* raise no issues */
+                    },
```
Comment (ddgenome) [inline index.ts]: I'm not crazy about the newline here.

Comment (cdupuis) [reply index.ts]: I removed it.

**Pull request merged**